src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -3510,36 +3510,47 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
+                        try
+                        {
+                            ProviderInfo targetProvider = null;
+                            CmdletProvider targetProviderInstance = null;
 
-                        var globbedTarget = Globber.GetGlobbedProviderPathsFromMonadPath(
-                            targetPath,
-                            allowNonexistingPath,
-                            context,
-                            out targetProvider,
-                            out targetProviderInstance);
+                            var globbedTarget = Globber.GetGlobbedProviderPathsFromMonadPath(
+                                targetPath,
+                                allowNonexistingPath,
+                                context,
+                                out targetProvider,
+                                out targetProviderInstance);
 
-                        if (!string.Equals(targetProvider.Name, "filesystem", StringComparison.OrdinalIgnoreCase))
-                        {
-                            throw PSTraceSource.NewNotSupportedException(SessionStateStrings.MustBeFileSystemPath);
-                        }
+                            if (!string.Equals(targetProvider.Name, "filesystem", StringComparison.OrdinalIgnoreCase))
+                            {
+                                throw PSTraceSource.NewNotSupportedException(SessionStateStrings.MustBeFileSystemPath);
+                            }
 
-                        if (globbedTarget.Count > 1)
-                        {
-                            throw PSTraceSource.NewInvalidOperationException(SessionStateStrings.PathResolvedToMultiple, targetPath);
-                        }
+                            if (globbedTarget.Count > 1)
+                            {
+                                throw PSTraceSource.NewInvalidOperationException(SessionStateStrings.PathResolvedToMultiple, targetPath);
+                            }
 
-                        if (globbedTarget.Count == 0)
-                        {
-                            throw PSTraceSource.NewInvalidOperationException(SessionStateStrings.PathNotFound, targetPath);
-                        }
+                            if (globbedTarget.Count == 0)
+                            {
+                                throw PSTraceSource.NewInvalidOperationException(SessionStateStrings.PathNotFound, targetPath);
+                            }
 
-                        // If the original target was a relative path, we want to leave it as relative if it did not require
-                        // globbing to resolve.
-                        if (WildcardPattern.ContainsWildcardCharacters(targetPath))
+                            // If the original target was a relative path, we want to leave it as relative if it did not require
+                            // globbing to resolve.
+                            if (WildcardPattern.ContainsWildcardCharacters(targetPath))
```

Comment:
This doesn't help, because the relative `content` path is ultimately still normalized in https://github.com/PowerShell/PowerShell/blob/90c7ed615c467347b42fe741a479cc168bf8fb38/src/System.Management.Automation/namespaces/FileSystemProvider.cs#L2379 - at least when I debug on macOS. Note that the relative path mustn't be used as-is: you must in effect perform the equivalent of `$relativePath -replace '[\\|]', [IO.Path]::DirectorySeparatorChar` to normalize the separators.

Reply:
Also, on a general note: I don't think it makes sense to ever interpret the target path as a wildcard pattern - see #13136

Reply:
Do you say about existent targets? In the case it is another issue. I don't change the behavior in the PR.  I only address non-existent targets.

Reply:
Yes, this is about existing targets (too), as described in #13064, which your PR description notes as targeting (in addition to #9067); also, the need to normalize the path separators in relative paths applies to both existing and non-existing targets.

I'd say it makes sense to fix all these things together, but if you don't want to do that, please at least remove #13064 from the PR description.

Reply:
I do not think that we can normalize non-existing targets safely. We can get unexpected results and can not check the targets.

Reply:
> I'd say it makes sense to fix all these things together

Can you please create a test I can use for the case you ask to fix? (All current tests are passed.)

Reply:
Since we are talking about _file-system_ paths, I do think we can normalize them safely - why you do think we can't?

Reply:
See tests below.

Current behavior (without your PR applied, observed on Windows, macOS, Ubuntu):

* Separator normalization already _is_ happening, namely with _non-existing_ targets.
* A non-existent target is currently accepted even _without_ `-Force` (a breaking change).
* On Windows only, the first test succeeds, because the relative path _is_ preserved - both for non-existent and existent targets - and, as stated separator normalization already happens; on macOS and Ubuntu - where relative paths are unexpectedly converted to full ones - both tests fails.

```powershell
Describe "New-Item symlinks" {
  BeforeAll {
    Push-Location testdrive:

    $lnks = "$PID-L1.tmp", "$PID-L2.tmp"

    # Same target, different separator syntax.
    $existingTargets = "./$PID.tmp", ".\$PID.tmp"
    $null > $existingTargets[0]

    $nonExistentTargets = './nosuch', '.\nosuch'

  }

  It "Can create symlink <lnk> with relative existing target <target>" -TestCases @{ lnk = $lnks[0]; target = $existingTargets[0] }, @{ lnk = $lnks[1]; target = $existingTargets[1] } {
    param([string] $lnk, [string] $target)

    $null = New-Item -Type SymbolicLink  $lnk -Target $target

    # Make sure the relative target path was recorded as such and uses
    # the platform-appropriate separator.
    (Get-Item $lnk).Target | Should -Be ($target -replace '[\\/]', [IO.Path]::DirectorySeparatorChar)

  }

  It "Can create symlink <lnk> with relative NON-existing target <target> with -Force" -TestCases @{ lnk = $lnks[0]; target = $nonExistentTargets[0] }, @{ lnk = $lnks[1]; target = $nonExistentTargets[1] } {
    param([string] $lnk, [string] $target)

    # Without -Force, this should fail (TODO: add specific error ID)
    { New-Item -ea Stop -Type SymbolicLink  $lnk -Target $target } | Should -Throw

    # Force creation.
    $null = New-Item -Type SymbolicLink  $lnk -Target $target -Force

    # Make sure the relative target path was recorded as such and uses
    # the platform-appropriate separator.
    (Get-Item $lnk).Target | Should -Be ($target -replace '[\\/]', [IO.Path]::DirectorySeparatorChar)

  }

  AfterAll {
    Pop-Location
  }

}
```

Reply:
@mklement0 Since all three issues (#9067,#13064,#13136) is based on the same code I am trying to fix all in the PR.

There is a problem with Target property on Unix-s. We call FollowSymLink() that is a native code from PowerShell-Native (libpsl-native)
https://github.com/PowerShell/PowerShell-Native/blob/7b7b889e8a331fd0ea47082d5edbf80402ea51ad/src/libpsl-native/src/followsymlink.cpp#L39-L47

[Docs explicitly](https://man7.org/linux/man-pages/man3/realpath.3.html) say that `realpath()` always returns _absolute_ path. I can not fix this part of #13064 in the PR.

Could you please check and confirm that PowerShell with the PR creates relative symlink correctly but Target is resolved to absolute path on Unix-s? If so we could close #13064 by the PR (with pending one test) and then open new issue for Target because psl is involved.

Reply:
Thanks, but I don't understand why `realpath()` returning an absolute path is a problem, given that you only need to use that path to _verify the target's existence_.

By contrast, the path to use for _creation of the link_ should be the one _as specified_, except for separator normalization.

Reply:
>  I don't understand why realpath() returning an absolute path is a problem

It is only used for Target property creation. Even if the symlink is relative, we will see the absolute path in dir output. I ask you to download the artifact and test - do you see that the symlink is really relative?

Reply:
I'm still confused: I also don't see why the `.Target` property shouldn't just use the target path _as defined_.
What reason is there to try to interpret it in any way?

Reply:
It seems we say about different things.
Currently if we create a symlink with relative path the Target property will return absolute path but we expect a relative path used at creation time. Because of this one test fails in the PR on Unix-s.

Reply:
I pulled your PR, built it on macOS 10.15 and Ubuntu 18.04 and can confirm that it correctly creates symlinks with relative paths and performs separator normalization (and that `.Target` always reports the _full_ path, assuming the target exists).

I'm still a bit confused about the `.Target` property: on Windows, it seemingly reports the target path _as stored in the link_. Why did we ever resolve it to a full path on Unix?

So your plan is to later change that and make it behave like it does on Windows?
But then you wouldn't need a call to `realpath()` at all, right?

Separately, knowing the full, resolved path is _also_ useful, so we could consider adding a `.ResolvedTarget` property that always reports the full path, if it exists. For non-symlink items, it could report that item's _own_ full path, which would enable you to target items uniformly by their ultimate target path.

Reply:
@mklement0 Thanks!

> So your plan is to later change that and make it behave like it does on Windows?
But then you wouldn't need a call to realpath() at all, right?

Yes, we should open new issue to track this. 

> Separately, knowing the full, resolved path is also useful

Please open new issue to discuss.

Reply:
See #13365 and #13366.